utils_linux.go

```diff
@@ -217,8 +218,11 @@ type runner struct {
 }
 
 func (r *runner) run(config *specs.Process) (_ int, retErr error) {
+	detach := r.detach || (r.action == CT_ACT_CREATE)
 	defer func() {
-		if retErr != nil {
+		// For a non-detached container, or we get an error, we
+		// should destroy the container.
+		if !detach || retErr != nil {
```

Comment:
nit: I think it is clearer to not put in a defer call the regular flow of a function. I'd keep the `!detach()` case in the main function, not in the defer.

Or is there a downside that I'm not seeing?

If you feel strongly about this, it's okay to keep it as it is. But if you doubt, it seems more readable for me to remove it.

Reply:
I disagree that a defer function is a clear way to communicate a main operation like killing a non-detached container. But okay.